docker/src/main/java/brooklyn/networking/sdn/ibm/SdnVeNetworkImpl.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2014 by Cloudsoft Corporation Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package brooklyn.networking.sdn.ibm;
+
+import java.util.Collection;
+
+import org.jclouds.net.domain.IpPermission;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import brooklyn.entity.Entity;
+import brooklyn.entity.basic.Entities;
+import brooklyn.entity.container.docker.DockerHost;
+import brooklyn.entity.proxying.EntitySpec;
+import brooklyn.location.basic.SshMachineLocation;
+import brooklyn.networking.sdn.SdnAgent;
+import brooklyn.networking.sdn.SdnProvider;
+import brooklyn.networking.sdn.SdnProviderImpl;
+import brooklyn.util.collections.MutableList;
+
+import com.google.common.collect.ImmutableList;
+
+public class SdnVeNetworkImpl extends SdnProviderImpl implements SdnVeNetwork {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SdnProvider.class);
+
+    @Override
+    public void init() {
+        LOG.info("Starting IBM SDN VE network id {}", getId());
+        super.init();
+
+        EntitySpec<?> agentSpec = EntitySpec.create(getConfig(SdnProvider.SDN_AGENT_SPEC, EntitySpec.create(SdnVeAgent.class)))
+                .configure(SdnVeAgent.SDN_PROVIDER, this);
+
+        setAttribute(SdnProvider.SDN_AGENT_SPEC, agentSpec);
+    }
+
+    @Override
+    public Collection<IpPermission> getIpPermissions() {
+        Collection<IpPermission> permissions = MutableList.of();
+        return permissions;
+    }
+
+    public void addHost(Entity item) {
+        SshMachineLocation machine = ((DockerHost) item).getDynamicLocation().getMachine();
+        EntitySpec<?> spec = EntitySpec.create(getAttribute(SDN_AGENT_SPEC))
+                .configure(SdnVeAgent.DOCKER_HOST, (DockerHost) item);
+        SdnVeAgent agent = (SdnVeAgent) getAgents().addChild(spec);
```

Comment:
Yes, the `Group` entity doesn't know the types of its members, they just have the `Entity` interface.